packages/dds/tree/src/test/shared-tree/sharedTree.bench.ts

```diff
@@ -0,0 +1,877 @@
+ * @param tree - tree in form of a Jsonable object
+ * @param shape - shape of the tree (wide vs deep)
+ */
+function manipulateTreeAsJSObject(tree: Jsonable, shape: TreeShape): void {
+	let nodesUnderRoot;
+	switch (shape) {
+		case TreeShape.Deep:
+			tree[0].value = 123;
+			break;
+		case TreeShape.Wide:
+			nodesUnderRoot = tree.fields.foo.length;
+			if (nodesUnderRoot === 0) {
+				tree.fields.value = 123;
+			} else {
+				tree.fields.foo[nodesUnderRoot - 1].value = 123;
+			}
+			break;
+		default:
+			unreachableCase(shape);
+	}
+}
+
+function getLeafNodeFromJSObject(tree: Jsonable): Jsonable {
+	for (const key of Object.keys(tree)) {
+		if (typeof tree[key] === "object" && tree[key] !== null) {
+			if (tree[key].type !== undefined && tree[key].fields === undefined) {
+				return tree;
+			}
+			return getLeafNodeFromJSObject(tree[key]);
+		}
+	}
+}
+
+function readCursorTree(forest: IForestSubscription, numberOfNodes: number, shape: TreeShape) {
+	const readCursor = forest.allocateCursor();
+	moveToDetachedField(forest, readCursor);
+	assert(readCursor.firstNode());
+	switch (shape) {
+		case TreeShape.Deep:
+			for (let i = 0; i < numberOfNodes; i++) {
+				readCursor.firstField();
+				assert(readCursor.firstNode());
+			}
+			break;
+		case TreeShape.Wide:
+			if (numberOfNodes === 1) {
+				break;
+			}
+			readCursor.firstField();
+			readCursor.firstNode();
+			for (let j = 0; j < numberOfNodes; j++) {
+				readCursor.nextNode();
+			}
+			break;
+		default:
+			unreachableCase(shape);
+	}
+	readCursor.free();
+}
+
+/**
+ * Given a tree and path, changes the node value to a different one.
+ * @param tree - tree that you need to manipulate
+ * @param path - location where you need to apply the edit
+ * @param dataType - the primitive data type of the value to store
+ */
+function manipulateCursorTree(tree: ISharedTree, path: UpPath, dataType: TestPrimitives) {
+	const value = generateTreeData(dataType);
+	tree.runTransaction((forest, editor) => {
+		editor.setValue(path, { type: brand("Test"), value });
+		return TransactionResult.Apply;
+	});
+}
+
+function getCursorLeafNode(numberOfNodes: number, shape: TreeShape): UpPath {
+	let path: UpPath;
+	switch (shape) {
+		case TreeShape.Deep:
+			path = {
+				parent: undefined,
+				parentField: rootFieldKeySymbol,
+				parentIndex: 0,
+			};
+			for (let i = 0; i < numberOfNodes; i++) {
+				path = {
+					parent: path,
+					parentField: localFieldKey,
+					parentIndex: 0,
+				};
+			}
+			return path;
+		case TreeShape.Wide:
+			path = {
+				parent: {
+					parent: undefined,
+					parentField: rootFieldKeySymbol,
+					parentIndex: 0,
+				},
+				parentField: localFieldKey,
+				parentIndex: numberOfNodes - 1,
+			};
+			return path;
+		default:
+			unreachableCase(shape);
+	}
+}
+
+function readEditableTree(tree: ISharedTree, numberOfNodes: number, shape: TreeShape) {
+	assert(isUnwrappedNode(tree.root));
+	let currField;
+	let currNode;
```

Comment:
Unfortunately if I try to declare these within the case block, I get a `Unexpected lexical declaration in case block.` complaint. is it okay to just disable this lint eslint error?

Reply:
The documentation for linter rule error usally provides guides for what to do. For example in this case, you need more braces to make the scoping work correctly, see https://eslint.org/docs/latest/rules/no-case-declarations (I just googled the error you listed to find that, buf when you are working in the code, the intelisense for the error should have a link)

Reply:
Also, always make sure you understand why a lint rule exists before you consider suppressing it: most of them exist because there is some unexpected issue thats likley to happen which you should understand, then document how that doesn't break your particular case if suppressing the rule.

Reply:
declared within the case block